### PR TITLE
Add support of sftp

### PR DESCRIPTION
**Pull request by ch0k0bn**
- pureftp provisioning is kept but not used atm

### CODE REVIEW

File: playbooks/group_vars/brain.yml [hunk 1]
```diff
@@ -97,30 +97,58 @@ rabbitmq_users_definitions:
...
-  PassivePortRange: "65435 65535"
-  # NOTE: vulnerable to POODLE but cannot find a workaround to make
-  # Python's 2.7.9 ftplib module work with Debian 8.1 pure-ftpd configured with
-  # TLSv1 only (-S:HIGH:MEDIUM:+TLSv1:!SSLv2:!SSLv3)
-  TLSCipherSuite: "-S:HIGH:MEDIUM:+TLSv1"
-
+# pureftpd_options:
+#  CreateHomeDir: "yes"
+#  PAMAuthentication: "no"
+# TLS: "2"
```
Comment (guillaumededrie) [inline playbooks/group_vars/brain.yml]: There is some missing spaces to align the next lines with previous one.

File: playbooks/group_vars/brain.yml [hunk 2]
```diff
@@ -97,30 +97,58 @@ rabbitmq_users_definitions:
...
+# PassivePortRange: "65435 65535"
+# # NOTE: vulnerable to POODLE but cannot find a workaround to make
+# # Python's 2.7.9 ftplib module work with Debian 8.1 pure-ftpd configured with
+# # TLSv1 only (-S:HIGH:MEDIUM:+TLSv1:!SSLv2:!SSLv3)
+# TLSCipherSuite: "-S:HIGH:MEDIUM:+TLSv1"
+
+## SSHD SFTP server
+sshd_sftp_server: internal-sftp
+sshd_match:
+  - Condition: "User {{irma_configurations.ftpd.frontend.username}}"
```
Comment (guillaumededrie) [inline playbooks/group_vars/brain.yml]: wow, nice! :+1:

File: playbooks/group_vars/brain.yml [hunk 3]
```diff
@@ -97,30 +97,58 @@ rabbitmq_users_definitions:
...
+    AllowTcpForwarding: no
+    PermitTunnel: no
+    ChrootDirectory: "{{ brain_sftpd_rootdir }}"
+    ForceCommand: "{{ sshd_sftp_server }} -u 2"
+
+## Brain role PureFtpd
+#brain_pureftpd_virtual_users:
+#  - username: "{{irma_configurations.ftpd.frontend.username}}"
+#    virtual_user: "ftpuser"
+#   virtual_home: "/home/ftpuser/frontend"
```
Comment (guillaumededrie) [inline playbooks/group_vars/brain.yml]: There is some missing spaces to align the next lines with previous one.

File: roles/quarkslab.irma_provisioning_brain/tasks/sftpd.yml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+- name: SFTPD | Ensure virtual users exists
+  user:
+    name: "{{ item.username }}"
+    group: "{{ brain_sftpd_groupname }}"
+    shell: /sbin/nologin
+    state: present
+  with_items: brain_sftpd_virtual_users
+
+- name: "SFTPD | Set sftp users password"
```
Comment (guillaumededrie) [inline roles/quarkslab.irma_provisioning_brain/tasks/sftpd.yml]: @ch0k0bn, is it possible to use the Ansible module `user` that is just before to set the user password?

Comment (ch0k0bn) [inline roles/quarkslab.irma_provisioning_brain/tasks/sftpd.yml]: i looked for using it but user role take only crypted password and after few failed attempts i try another approach to keep clear password.

Comment (guillaumededrie) [inline roles/quarkslab.irma_provisioning_brain/tasks/sftpd.yml]: Ok, got it, maybe in the future we can use the `password lookup` module.

**Pull request merged**